src/middleware/authorisation.coffee

```diff
@@ -75,10 +75,16 @@ exports.authorise = (ctx, done) ->
       pat = new RegExp channel.urlPattern
       # if url pattern matches
       if pat.test ctx.request.url
-        matchedRoles = channel.allow.filter (element) ->
-          return (ctx.authenticated.roles.indexOf element) isnt -1
+        matchedRoles = {}
+        if ctx.authenticated?
+          if ctx.authenticated.roles?
+            matchedRoles = channel.allow.filter (element) ->
+              return (ctx.authenticated.roles.indexOf element) isnt -1
+        else
+          ctx.authenticated = {}
```

Comment:
The messageStore middleware depends on having ctx.authenticated being set to something other than null, otherwise it throws an error, I have since fixed this in a later commit.